src/Deinterleave.cpp

```diff
@@ -402,6 +407,12 @@ Expr deinterleave(Expr e, int starting_lane, int lane_stride, int new_lanes, con
     Deinterleaver d(starting_lane, lane_stride, new_lanes, lets);
     e = d.mutate(e);
     e = common_subexpression_elimination(e);
+    if (const Shuffle *shuffle = e.as<Shuffle>()) {
+        if (shuffle->is_extract_element() && shuffle->vectors.size() == 1) {
```

Comment:
Remove this if, and just don't simplify if the root node is a shuffle. Add a comment about simplify calling this function on shuffles to explain why.